subsys/pm/policy/policy_residency.c

```diff
@@ -33,37 +33,30 @@ static int states_per_cpu[CONFIG_MP_NUM_CPUS] = {};
 
 struct pm_state_info pm_policy_next_state(uint8_t cpu, int32_t ticks)
 {
-	int i;
-
 	CHECKIF(cpu >= ARRAY_SIZE(states_per_cpu)) {
 		goto error;
 	}
 
-	i = states_per_cpu[cpu];
-	const struct pm_state_info *states = (const struct pm_state_info *)
-		cpus_states[cpu];
-
-	for (i = i - 1; i >= 0; i--) {
+	for (uint8_t i = states_per_cpu[cpu] - 1U; i >= 0U; i--) {
```

Comment:
You have to keep it signed, otherwise it will have an underflow and the loop will continue.

Reply:
right, will use `int16_t` here.